### ISSUE TITLE
bug:  type library does not correctly identify nullable for complex types

### ISSUE THREAD

**Issue by EpsilonPrime** [substrait-io/substrait-cpp#52]
Example:
```
  testDecode<ParameterizedList>(
      "list<list<string?>>",
      [](const std::shared_ptr<const ParameterizedList>& typePtr) {
        ASSERT_FALSE(typePtr->nullable());  // returns true
        ASSERT_FALSE(typePtr->elementType()->nullable());  // returns true
        ASSERT_TRUE(std::dynamic_pointer_cast<const ParameterizedList>(
                        typePtr->elementType())
                        ->elementType()
                        ->nullable());
        ASSERT_EQ(typePtr->signature(), "list<list<str>>");
      });
```

**Comment by EpsilonPrime** [comment]
Addressed in #53.

**Comment by EpsilonPrime** [comment]
#53 has been merged.
